crypto/src/e2e_identity/mod.rs

```diff
@@ -81,6 +81,16 @@ impl MlsCentral {
         let sk = enrollment.get_sign_key_for_mls()?;
         let cs = enrollment.ciphersuite;
         let certificate_chain = enrollment.certificate_response(certificate_chain).await?;
+
+        // Own intermediates are not provided by smallstep in the /federation endpoint so we got to intercept them here, at issuance
+        let size = certificate_chain.len();
+        if size > 3 {
+            let intermediates = &certificate_chain[1..size - 1];
+            for int in intermediates {
+                self.e2ei_register_intermediate_ca_der(int).await?
+            }
```

Comment:
You probably need to register them in *the reverse order* you're seeing them in the list; 
i.e. here you have [ee, inter3, inter2, inter1]. In the case any of inter2 and 3 are actually signed by inter1, you need to have registered inter1 beforehand or inter2 or 3 won't validate.